nsv13/code/modules/overmap/weapons/ship_weapon.dm

```diff
@@ -1,10 +1,15 @@
 #define FIRE_INTERCEPTED 2 //For special_fire()
 
+//These procs should *really* not be here
 /obj/structure/overmap/proc/add_weapon(obj/machinery/ship_weapon/weapon)
 	if(weapon_types[weapon.fire_mode])
 		var/datum/ship_weapon/SW = weapon_types[weapon.fire_mode]
 		SW.add_weapon(weapon) //hand it over to the datum for sane things like adding it idk
 
+/obj/structure/overmap/proc/remove_weapon(obj/machinery/ship_weapon/weapon)
+	var/datum/ship_weapon/SW = weapon_types[weapon.fire_mode]
+	SW.weapons -= weapon
```

Comment:
weapons is a list with sublists so this won't do, remove it from the sublists instead
( SW.weapons["all"] -= weapon and SW.weapons["loaded"] -= weapon )

Also the proc is unused, you may want to add it to where you wanted it to be.

Reply:
Whoops, probably good to use it yeah!